pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java

```diff
@@ -280,6 +280,7 @@ public static class QueryOptionKey {
         public static final String SKIP_UPSERT = "skipUpsert";
         public static final String USE_STAR_TREE = "useStarTree";
         public static final String SCAN_STAR_TREE_NODES = "scanStarTreeNodes";
+        public static final String SCAN_STAR_TREE_NODES_ON_HEAP = "scanStarTreeNodesOnHeap";
```

Comment:
Should we come up with one single query option for the data locality based optimizations?

Reply:
maybe just the `scanStarTreeNodes` option to indicate the reads are moving forward (which indicates the use of cache to avoid looking back)

Reply:
called it reduceRandomAccess to be more descriptive and generic.